examples_tests/61.BlitFilterTest/default_compute_normalization.comp

```diff
@@ -0,0 +1,192 @@
+
+#include <../normalization/parameters.glsl>
+#include <../normalization/normalization.glsl>
+#include <../normalization/descriptors.glsl>
+// #include <../blit/formats/rgba16.glsl>
+
+layout (push_constant) uniform Block
+{
+	nbl_glsl_blit_normalization_parameters_t params;
+} pc;
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_GET_PARAMETERS_DEFINED_
+nbl_glsl_blit_normalization_parameters_t nbl_glsl_blit_normalization_getParameters()
+{
+	return pc.params;
+}
+#define _NBL_GLSL_BLIT_NORMALIZATION_GET_PARAMETERS_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_GET_PADDED_DATA_DEFINED_
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_
+#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_ must be defined
+#endif
+
+nbl_glsl_blit_normalization_pixel_t nbl_glsl_blit_normalization_getPaddedData(in ivec3 coord)
+{
+	const nbl_glsl_blit_normalization_parameters_t params = nbl_glsl_blit_normalization_getParameters();
+
+#if NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 1)
+	#define COORD coord.x
+	#define IMAGE_DIM params.outImageDim.x
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 2)
+	#define COORD coord.xy
+	#define IMAGE_DIM params.outImageDim.xy
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 3)
+	#define COORD coord
+	#define IMAGE_DIM params.outImageDim
+#else
+	#error _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_ not supported
+#endif
+
+	nbl_glsl_blit_normalization_pixel_t result;
+	result.data = vec4(0.f);
+	if (all(lessThan(COORD, IMAGE_DIM)))
+	{
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		result.data = nbl_glsl_blit_formats_decode(imageLoad(_NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_, COORD));
+	#else
+		result.data = imageLoad(_NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_, COORD);
+	#endif
+	}
+
+#undef COORD
+#undef IMAGE_DIM
+
+	return result;
+}
+#define _NBL_GLSL_BLIT_NORMALIZATION_GET_PADDED_DATA_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_SET_DATA_DEFINED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_SET_DATA_DEFINED_
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_
+#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_ must be defined
+#endif
+
+void nbl_glsl_blit_normalization_setData(in nbl_glsl_blit_normalization_pixel_t value, in ivec3 coord)
+{
+	const nbl_glsl_blit_normalization_parameters_t params = nbl_glsl_blit_normalization_getParameters();
+
+#if NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 1)
+	#define COORD coord.x
+	#define IMAGE_DIM params.outImageDim.x
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 2)
+	#define COORD coord.xy
+	#define IMAGE_DIM params.outImageDim.xy
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 3)
+	#define COORD coord
+	#define IMAGE_DIM params.outImageDim
+#else
+	#error _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_ not supported
+#endif
+
+	if (all(lessThan(COORD, IMAGE_DIM)))
```

Comment:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/343/files#r864376210

Reply:
basically, hoist the check

Reply:
Resolved in the following commit.

Reply:
let me know when pushed

Reply:
Pushed.